tools/configure

```diff
@@ -1,291 +1,176 @@
-# srcroot is not used here
-my $srcroot = "";
-$config->add_config_variables($files_spec_file, $srcroot, $cimeroot, $model);
-# todo:
-#$config->set('MACHDIR', "$machdir");
-my $file = $config->get('CONFIG_DRV_FILE');
-
-$config->add_config_variables($file, $srcroot, $cimeroot, $model);
-
-ConfigMachine::setMachineValues($file, "", $mach, $config);
-
-
-# This will be the compiler argument passed in or the default if not provided
-$compiler = $config->get('COMPILER');
-
-$logger->info( "Machine specifier: $mach.$eol");
-
-# Copy Depends files if they exist
-if( -e "$machdir/Depends.$mach" ) {
-    unless(copy("$machdir/Depends.$mach",$output_dir)){
-	$logger->logdie( "ERROR:  copy $machdir/Depends.$mach $output_dir failed$eol");
-	exit;
-    }
-}
-if( -e "$machdir/Depends.$compiler" ) {
-    unless(copy("$machdir/Depends.$compiler",$output_dir)){
-	$logger->logdie( "ERROR:  copy $machdir/Depends.$compiler $output_dir failed$eol");
-    }
-}
-
-my $dbug = $config->get("DEBUG");
-
-my $moduleloader = Module::ModuleLoader->new(machine  => $mach,
-					     compiler => $compiler,
-					     mpilib   => $mpilib,
-					     caseroot => $output_dir,
-					     debug    => $dbug,
-					     cimeroot => $cimeroot,
-					     model    => $model);
-
-$moduleloader->moduleInit();
-$moduleloader->writeXMLFileForCase();
-
-
-
-$moduleloader->writeCshModuleFile();
-$moduleloader->writeShModuleFile();
-
-
-SetupTools::set_compiler($config->get('OS'),"$machdir/$compiler_file",$compiler,
-			 $mach, $mpilib, "$output_dir/Macros.$output_format", $output_format);
-
-$logger->info("Successfully created auxilary build files for $mach $eol");
-
-exit 0;
-
-
+#!/usr/bin/env python
+
+"""This script writes CIME build information to a directory.
+
+The pieces of information that will be written include:
+
+1. Machine-specific build settings (i.e. the "Macros" file).
+2. File-specific build settings (i.e. "Depends" files).
+3. Environment variable loads (i.e. the env_mach_specific files).
+
+The .env_mach_specific.sh and .env_mach_specific.csh files are specific to a
+given compiler, MPI library, and DEBUG setting. By default, these will be the
+machine's default compiler, the machine's default MPI library, and FALSE,
+respectively. These can be changed by setting the environment variables
+COMPILER, MPILIB, and DEBUG, respectively.
+"""
+
+import os
+import shutil
+import sys
+
+_CIMEROOT = os.environ.get("CIMEROOT")
+if _CIMEROOT is None:
+    raise SystemExit("ERROR: must set CIMEROOT environment variable")
+
+_LIBDIR = os.path.join(_CIMEROOT, "scripts", "Tools")
+sys.path.append(_LIBDIR)
+
+from standard_script_setup import *
+from CIME.utils import expect
+from CIME.XML.build import Build
+from CIME.XML.env_mach_specific import EnvMachSpecific
+from CIME.XML.machines import Machines
+
+logger = logging.getLogger(__name__)
+
+def parse_command_line(args):
+    """Command line argument parser for configure."""
+    description = __doc__
```

Comment:
I did test this. The `__doc__` here does indeed refer to the global definition. The docstring for `parse_command_line` is stored as an attribute of the function, i.e. as `parse_command_line.__doc__`, so it is *not* a local variable. TLDR; there's no name clash here, everything is fine. :smiley: